src/services/supabase.ts

```diff
@@ -278,6 +227,134 @@ export const jobSucceeded = (jobStatus?: JobStatus) => {
     }
 };
 
+// START: Poller
+type PollerTimeout = number | undefined;
+const INTERVAL_MAX = 5000;
+const INTERVAL_INCREMENT = 500;
+export const JOB_STATUS_POLLER_ERROR = 'supabase.poller.failed';
+const cleanUp = (pollerTimeout: PollerTimeout) => {
+    if (pollerTimeout) {
+        window.clearInterval(pollerTimeout);
+    }
+};
+
+export const jobStatusPoller = (
+    query: any,
+    success: Function,
+    failure: Function,
+    initWait?: number
+) => {
+    console.log('jobStatusPoller');
+    let pollerTimeout: PollerTimeout;
+    let interval = DEFAULT_POLLING_INTERVAL;
+    const makeApiCall = () => {
+        LogRocket.log('Poller : start ');
+
+        return query.throwOnError().then(
+            (payload: any) => {
+                LogRocket.log('Poller : response : ', payload);
+                cleanUp(pollerTimeout);
+
+                if (payload.error) {
+                    failure(handleFailure(payload.error));
+                } else {
+                    const response =
+                        (payload &&
+                            hasLength(payload.data) &&
+                            payload.data[0]) ??
+                        null;
+                    if (
+                        response?.job_status?.type &&
+                        response.job_status.type !== 'queued'
+                    ) {
+                        LogRocket.log(
+                            `Poller : response : ${response.job_status.type}`
+                        );
+                        if (response.job_status.type === 'success') {
+                            success(response);
+                        } else {
+                            failure(response);
+                        }
+                    } else {
+                        interval =
+                            interval < INTERVAL_MAX
+                                ? interval + INTERVAL_INCREMENT
+                                : INTERVAL_MAX;
+                        pollerTimeout = window.setTimeout(
+                            makeApiCall,
+                            interval
+                        );
+                    }
```

Comment:
Most of this is identical to the subscription handler.

However we not need a block to handle calculating the interval.